### PR TITLE
core-analytics@1.21.0 - Clone of f-analytics without dependency on Vue, Vuex & SSR

### PR DESCRIPTION
**Pull request by oliversweb**
### Changed (**breaking changes**)
- Cloned the original `f-analytics` into `core-analytics` and remove any refs. to Vue/Vuex and any SSR functionality.

### CODE REVIEW

File: packages/services/core-analytics/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+dist
+harness
```
Comment (webbertakken) [inline packages/services/core-analytics/.eslintignore]: Can't find any ref to harness. Is it used?

File: packages/services/core-analytics/LICENSE [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Apache License
+Version 2.0, January 2004
+http://www.apache.org/licenses/
+
+Copyright (c) Just Eat Holding Ltd
```
Comment (webbertakken) [inline packages/services/core-analytics/LICENSE]: Copyright should be to `Just Eat Takeaway.com`

**Review by webbertakken — DISMISSED** [review]
LGTM

File: packages/services/core-analytics/.eslintignore (hunk 1 above)
Comment (oliversweb) [reply packages/services/core-analytics/.eslintignore]: Good spot, I took that folder out, I will remove

**Review by webbertakken — DISMISSED** [review]

File: packages/services/core-analytics/src/tests/helpers/setup.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+};
+
+const createStore = ({
+    name = options.namespace,
+    state = {},
+    actions = defaultActions,
+    getters = defaultGetters,
+    mutations = defaultMutations
+} = {}) => {
+    Vue.use(Vuex);
```
Comment (tommcclean) [inline packages/services/core-analytics/src/tests/helpers/setup.js]: Should these bits still be here?

File: packages/services/core-analytics/src/services/analytics.mapper.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+export const mapUserData = ({ userData, authToken } = {}) => {
+    const userId = mapAnonymousUserId();
+    let mappedUserData = {
+        ...userData,
+        'a-UserId': userId || userData['a-UserId']
+    };
+
+    if (authToken) {
+        const tokenData = jwtDecode(authToken);
+        const authType = tokenData?.is_new_registration ? GRANT_TYPES.registration : (GRANT_TYPES[tokenData?.grant_type] || GRANT_TYPES.default);
```
Comment (tommcclean) [inline packages/services/core-analytics/src/services/analytics.mapper.js:93]: ℹ️ - if we use this in non JE things, the claims might not be identical(?); just one to bear in mind

File: packages/services/core-analytics/src/services/analytics.mapper.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    return value;
+};
+
+/**
+ * Maps the anonymous User Id to the UserData 'a-UserId' field.
+ *
+ * @return {string} UserId from je-auser cookie
+ */
+const mapAnonymousUserId = () => {
+    const value = getCookie('je-auser');
```
Comment (tommcclean) [inline packages/services/core-analytics/src/services/analytics.mapper.js]: Perhaps drive this cookie name from config (maybe next pr?) as I think we call it `jet-auser` in certain codebases

File: packages/services/core-analytics/README.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+## Usage
+
+*  <strong>Install the module using npm or Yarn</strong>
+    ```
+    yarn add @justeat/core-analytics
+    ```
+    ```
+    npm install @justeat/core-analytics
+    ```
```
Comment (webbertakken) [inline packages/services/core-analytics/README.md:44]: Recommend putting description and commands in the same order

File: packages/services/core-analytics/README.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+          ...
+
+          window.onload = function() {
+              this.service.pushPlatformData();
+          };
+
+          ...
+
+      </script>
+      ```
```
Comment (webbertakken) [inline packages/services/core-analytics/README.md:98]: codeblock is html, not js

Comment (webbertakken) [reply packages/services/core-analytics/README.md:98]: Same for a few other codeblocks down here.

File: packages/services/core-analytics/package.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    "engines": {
+        "node": "^12 || ^14 || ^16"
+    },
+    "keywords": [
+        "fozzie"
+    ],
+    "scripts": {
+        "prepublishOnly": "yarn lint && yarn test && yarn build",
+        "build": "vite build",
+        "lint": "eslint \"!(dist)/**/*.{js,vue}\"",
```
Comment (webbertakken) [inline packages/services/core-analytics/package.json:36]: probably better to `.eslintignore` the `dist/` folder

**Review by webbertakken — APPROVED** [review]
Nice work!

**Review by tommcclean — APPROVED** [review]

**Pull request merged**